### PR TITLE
Upgrade to etcd server 3.3.17

### PR DESCRIPTION
**Pull request by jpbetz**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

In https://github.com/kubernetes/kubernetes/pull/83801 we upgraded the etcd client to 3.3.17. While there are no server side changes between etcd 3.3.15 and 3.3.17, we're upgrading it for consistency.

```release-note
Upgrade default etcd server version to 3.3.17
```

/sig api-machinery
/area etcd
/priority critical-urgent

### CODE REVIEW

**Comment by jpbetz** [comment]
/hold
Until #83801 merged and has had sufficient bake time.

**Comment by aojea** [comment]
it's failing with this error

> W1011 21:11:29.589] Failed to pull k8s.gcr.io/etcd:3.3.17-0 with error: command "docker pull k8s.gcr.io/etcd:3.3.17-0" failed with error: exit status 1

**Comment by neolit123** [comment]
> is the image available in the registry?

not yet. i will be added later.

**Comment by neolit123** [comment]
thanks @jpbetz 
/approve

**Comment by jpbetz** [comment]
Image should be available. I’ll review CI failures tonight

**Review by odinuge — APPROVED** [review]
There are still references to `v3.3.15` on line 200 in [hack/lib/etcd.sh](https://github.com/jpbetz/kubernetes/blob/etcd-3_3_17_server/hack/lib/etcd.sh#L19) (and [here](https://github.com/jpbetz/kubernetes/blob/etcd-3_3_17_server/cluster/images/etcd/README.md#usage) and [here](https://github.com/jpbetz/kubernetes/blob/etcd-3_3_17_server/cluster/gce/config-test.sh#L202) as examples).

Adding a new test for the [etcd version for k8s v1.17](https://github.com/jpbetz/kubernetes/blob/etcd-3_3_17_server/cmd/kubeadm/app/images/images_test.go#L98) wouldn't hurt either I guess. Otherwise lgtm.

/lgtm

**Comment by jpbetz** [comment]
`./hack/verify-external-dependencies-version.sh` now passes.

**Comment by jpbetz** [comment]
/hold cancel
#83801 is merged

**Comment by wojtek-t** [comment]
/approve

Thanks Joe!

**Pull request merged**
